### DISCUSSION TITLE
Access Windows OneDrive directory from Linux

### DISCUSSION THREAD

**Discussion by KarlNapp2023** [abraunegg/onedrive#2531] [General]
Dear all,

I try to use the same directory for syncing to OneDrive from both Windows 10 and Linux Mint 21.2 (dual boot system). The OneDrive directory is on a NTFS SSD.

Unfortunately I can't access the directory from Linux if the OneDrive directory is connected to the Microsoft account (i.e. syncing is active, existing reparse point). When I remove this connection, I can access the OneDrive directory from Linux (reparse point gone).

Error message in Linux when trying to access the OneDrive directory: "unsupported reparse tag 0x9000701"

From my understanding this behaviour is caused by files on demand turned on. Unfortunately this feature is turned off and all files are saved on my NTFS SDD (almost 100 GB)

What didn't work:

1. Removing the reparse point via `fsutil reparsepoint delete D:\OneDrive` => access denied

2. Turning off files on demand. Unfortunately the screenshots in onedrive/docs/advanced-usage.md look different, so I pressed "Download all files" and checked "Always keep on this device"

![OneDrive](https://github.com/abraunegg/onedrive/assets/149235510/1ab5f713-dea4-4679-9a4c-9ea2f5536566)

3. Turning off files on demand in policies (gepedit.msc)

Any idea how to use the same OneDrive directory both in Windows and Linux?

**Comment by abraunegg** [comment]
@KarlNapp2023 

> Any idea how to use the same OneDrive directory both in Windows and Linux?

The *only* way for you to share the data between Windows and Linux is to *disable* Files On-Demand in Windows.

If this is becoming problematic to unset Files On-Demand in Windows - then this is a Windows problem that I cannot help you with.

Suggest you reach out to Microsoft and/or Windows support forums for assistance in disabling Files On-Demand in Windows.

**Comment by KarlNapp2023** [comment]
Thanks for the information. I finally managed to gain access to the folder by installing ntfs-3g-onedrive (just in case someone else has the same problem)

https://github.com/gbrielgustavo/ntfs-3g-onedrive/

**Reply by patatetom** [reply]
hi,
I'm trying to install this plugin and I'm stuck on a question :
do I have to recompile or can I use the binary in the `linux-64` folder directly ?
this one must date from 2021 and is based on (lib)ntfs-3G 2017.
regards, lacsaP.

**Reply by abraunegg** [reply]
@patatetom 
My advice is you should always recompile source for your platform.

**Reply by patatetom** [reply]
@abraunegg 
I'm going to go with this approach, which seems to me to be the most appropriate.
I'm not used to compiling and, when I run it following the instructions given, it generates two files `ntfs-plugin-9000001a.so` and `ntfs-plugin-9000001a.la` : is the second necessary for the first to work properly ?
according to the internet, the text file `ntfs-plugin-9000001a.la` is relative to `libtool` and is therefore only necessary/useful for compiling : is that correct ?
regards, lacsaP.

**Reply by abraunegg** [reply]
@patatetom 
I have no idea regarding that tool. Best raise a question on that GitHub repo for your answers - sorry.

**Reply by kantoulf** [reply]
Hi, I managed to install the plugin and got read, but not WRITE access to my windows Onedrive folder. Any solutions ? sudo chmod doesn't work here.

**Reply by patatetom** [reply]
hi,
I can't give you any information on this point, as I've installed this plugin in order to have RO access to any files that may be present.
have you checked that the mount (by hand) was done in RW mode and that the UID option is set to the right user (you) ?
regards, lacsaP.

**Reply by abraunegg** [reply]
@kantoulf 
It is best you ask your question regarding Read|Write access on https://github.com/gbrielgustavo/ntfs-3g-onedrive/ 

As @patatetom mentions - it could be to do with how you are mounting your Windows drive under Linux

**Comment by Sapus-Aphybius** [comment]
Oi gente, não sei se alguém conseguiu resolver essa questão. Mas atualmente o driver ntfs3 vem com suporte para reparse points do onedrive. Então, não precisa fazer nada, basta usar uma distribuićão que use esse driver no kernel. Estou usando e funciona normalmente, para ler e escrever na pasta do onedrive do windows acessando por dentro do linux.
